helpers/helpers.html

```diff
@@ -0,0 +1,102 @@
+    t /= d;
+    return -c * t*(t-2) + b;
+  };
+
+  /**
+   * Helper function that scrolls to a particular set of coordinates in a scroll target.
+   * If the scroll target is not defined, then it would use the main document as the target.
+   *
+   * To scroll in a smooth fashion, you can set the option `behavior: 'smooth'`. e.g.
+   *
+   * ```js
+   * Polymer.AppLayout.scroll({top: 0, behavior: 'smooth'});
+   * ```
+   *
+   * To scroll in a silent mode, without notifying scroll changes to any app-layout elements,
+   * you can set the option `behavior: 'silent'`. This is particularly useful we you are using
+   * `app-header` and you desire to scroll to the top of a scrolling region without running
+   * scroll effects. e.g.
+   *
+   * ```js
+   * Polymer.AppLayout.scroll({top: 0, behavior: 'silent'});
+   * ```
+   *
+   * @param {object} options {top: Number, left: Number, behavior: String(smooth | silent)}
+   */
+  Polymer.AppLayout.scroll = function scroll(options) {
+    options = options || {};
+
+    var docEl = document.documentElement;
+    var target = options.target || docEl;
+    var initScrollBehavior = target.style.scrollBehavior;
+    var scrollClassName = 'app-layout-silent-scroll';
+    var scrollTop = options.top || 0;
+    var scrollLeft = options.left || 0;
+    var smooth = options.behavior === 'smooth';
+    var silent = options.behavior === 'silent';
+
+    var scrollTo = target === docEl ? window.scrollTo :
+      function scrollTo(scrollLeft, scrollTop) {
+        target.scrollLeft = scrollLeft;
+        target.scrollTop = scrollTop;
+      };
+
+    if (smooth) {
+      if (initScrollBehavior == null) {
+        var timingFn = Polymer.AppLayout.scrollTimingFunction;
+        var startTime = Date.now();
+        var currentScrollTop = target === docEl ? window.pageYOffset : target.scrollTop;
+        var currentScrollLeft = target === docEl ? window.pageXOffset : target.scrollLeft;
+        var deltaScrollTop = scrollTop - currentScrollTop;
+        var deltaScrollLeft = scrollLeft - currentScrollLeft;
+        var duration = 300;
+
+        (function updateFrame() {
+          var now = Date.now();
+          var elapsedTime = now - startTime;
+
+          if (elapsedTime < duration) {
+            scrollTo(
+                timingFn(elapsedTime, currentScrollLeft, deltaScrollLeft, duration),
+                timingFn(elapsedTime, currentScrollTop, deltaScrollTop, duration)
+              );
+            requestAnimationFrame(updateFrame.bind(this));
+          }
+        }).call(this);
+
+      } else {
+        target.style.scrollBehavior = 'smooth';
+        target.offsetWidth && scrollTo(scrollLeft, scrollTop);
+        target.style.scrollBehavior = initScrollBehavior;
+      }
+    } else if (silent) {
+      docEl.classList.add(scrollClassName);
+      scrollTo(scrollLeft, scrollTop);
+
+      requestAnimationFrame(function() {
+        docEl.classList.remove(scrollClassName);
+      });
```

Comment:
The `silent` behavior doesn't seem to work.  Try running pesto, go to recipe page and scroll to the bottom of the page.  If you run `Polymer.AppLayout.scroll({top: 0, left: 0, behavior: 'silent'})` on the console you will see header bg transition happening.